extensions/spack-fairsoft/fairsoft/cmd/fairsoft.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+# Copyright 2020 GSI Helmholtz Centre for Heavy Ion Research GmbH, Darmstadt
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from llnl.util.tty.colify import colify
+import llnl.util.tty as tty
+import spack.extensions.fairsoft as ext
+import sys
+
+description = 'manage FairSoft distros'
+section = 'FairSoft distro'
+level = 'short'
+
+
+def avail(args):
+    """show available FairSoft distros"""
+    distros = sorted(ext.get_envs())
+
+    if sys.stdout.isatty():
+        if not distros:
+            tty.msg('No distros')
+        else:
+            tty.msg('%d distros' % len(distros))
+
+    colify(distros, indent=4, cols=1)
+
+
+def info(args):
+    """show info on a FairSoft distro"""
+    raise NotImplementedError('NOT YET IMPLEMENTED')
+
+
+def install(args):
+    """install a FairSoft distro"""
+    raise NotImplementedError('NOT YET IMPLEMENTED')
+
+
+def list(args):
+    """list installed FairSoft distros"""
+    raise NotImplementedError('NOT YET IMPLEMENTED')
+
+
+def setup(args):
+    """setup spack for FairSoft"""
+    raise NotImplementedError('NOT YET IMPLEMENTED')
+
+
+def view(args):
+    """manage symlink spack views on a FairSoft distro"""
+    raise NotImplementedError('NOT YET IMPLEMENTED')
+
+
+def setup_parser(parser):
+    subcmds = parser.add_subparsers(metavar='SUBCOMMAND',
+                                    dest='fairsoft_subcommand')
+
+    subcmds.add_parser('avail', help=avail.__doc__)
+    subcmds.add_parser('info', help=info.__doc__)
+    subcmds.add_parser('install', help=install.__doc__)
+    subcmds.add_parser('list', help=list.__doc__)
+    subcmds.add_parser('setup', help=setup.__doc__)
+    subcmds.add_parser('view', help=view.__doc__)
+
+
+def fairsoft(parser, args):
+    """Dispatch to SUBCOMMAND action"""
+    globals()[args.fairsoft_subcommand](args)
```

Comment:
I don't know, whether I like this.

Reply:
I am open for anything here. It is copied from how they do it inside spack itself. I read the python docs and it seems to be the proper way to inspect module-level functions.

Reply:
Technically, all is fine. I can't say explicitly, what makes me a bit nervous. If I know something better, I'll notify you.

Reply:
> I can't say explicitly, what makes me a bit nervous.

Maybe because it smells like one could call arbitrary functions, that was one of my concerns for a moment. But the parser actually verifies and denies arbritrary strings.

Anyhow, I have changed it to an explicit module-level actions dictionary - few more lines, but also more readable I find.